pandas/tests/generic/test_generic.py

```diff
@@ -592,6 +592,31 @@ def test_copy_and_deepcopy(self):
                 assert obj_copy is not obj
                 self._compare(obj_copy, obj)
 
+    @pytest.mark.parametrize("periods,fill_method,limit,exp", [
+        (1, "ffill", None, [np.nan, np.nan, np.nan, 1, 1, 1.5, 0, 0]),
+        (1, "ffill", 1, [np.nan, np.nan, np.nan, 1, 1, 1.5, 0, np.nan]),
+        (1, "bfill", None, [np.nan, 0, 0, 1, 1, 1.5, np.nan, np.nan]),
+        (1, "bfill", 1, [np.nan, np.nan, 0, 1, 1, 1.5, np.nan, np.nan]),
+        (-1, "ffill", None, [np.nan, np.nan, -.5, -.5, -.6, 0, 0, np.nan]),
+        (-1, "ffill", 1, [np.nan, np.nan, -.5, -.5, -.6, 0, np.nan, np.nan]),
+        (-1, "bfill", None, [0, 0, -.5, -.5, -.6, np.nan, np.nan, np.nan]),
+        (-1, "bfill", 1, [np.nan, 0, -.5, -.5, -.6, np.nan, np.nan, np.nan])
+    ])
+    def test_pct_change(self, periods, fill_method, limit, exp):
+        obj = self._construct(0)
+        is_frame = type(obj) is DataFrame
+        ser = pd.Series([np.nan, np.nan, 1, 2, 4, 10, np.nan, np.nan])
+        obj = pd.concat((obj, ser))
+
+        func = getattr(obj, 'pct_change')
+        res = func(periods=periods, fill_method=fill_method, limit=limit)
+
+        # Convert output into Series, regardless of input type
```

Comment:
if you rename obj above, you can simply determine the output down here, and make this a bit less convoluted, e.g.

```
if type(obj) is DataFrame:
    # compare dataframe
else:
   # compare series
```